lib/ssl/test/ssl_ocsp_SUITE.erl

```diff
@@ -0,0 +1,367 @@
+     {group, revoked_1_2},
+     {group, revoked_1_3}].
+
+groups() -> 
+    [{tls1_2, [], tls1_2_tests()},
+     {tls1_3, [], tls1_3_tests()},
+     {revoked_1_2, [], tls1_2_revoked_tests()},
+     {revoked_1_3, [], tls1_3_revoked_tests()}].
+
+tls1_2_tests() ->
+    [ocsp_stapling_without_nonce_and_responder_certs_tls1_2,
+     ocsp_stapling_with_nonce_tls1_2,
+     ocsp_stapling_with_responder_cert_tls1_2].
+
+tls1_2_revoked_tests() ->
+    [ocsp_stapling_revoked_tls1_2].
+
+tls1_3_tests() ->
+    [ocsp_stapling_without_nonce_and_responder_certs_tls1_3,
+     ocsp_stapling_with_nonce_and_responder_certs_tls1_3].
+
+tls1_3_revoked_tests() ->
+    [ocsp_stapling_revoked_tls1_3].
+
+%%--------------------------------------------------------------------
+init_per_suite(Config) ->
+    catch crypto:stop(),
+    try crypto:start() of
+	ok ->
+        ssl_test_lib:clean_start(),
+        DataDir = proplists:get_value(data_dir, Config),
+        PrivDir = proplists:get_value(priv_dir, Config),
+
+        %% Prepare certs
+        {ok, _} = make_certs:all(DataDir, PrivDir),
+
+        ResponderPort = get_free_port(),
+        Pid =
+            start_ocsp_responder(
+                erlang:integer_to_list(ResponderPort), PrivDir),
+
+        NewConfig =
+        lists:merge(
+            [{responder_port, ResponderPort},
+             {responder_pid, Pid}
+            ], Config),
+
+	    ssl_test_lib:cert_options(NewConfig)
+    catch _:_ ->
+	    {skip, "Crypto did not start"}
+    end.
+
+
+end_per_suite(Config) ->
+    ResponderPid = ?config(responder_pid, Config),
+    stop_ocsp_responder(ResponderPid),
+    ok = ssl:stop(),
+    application:stop(crypto).
+
+%%--------------------------------------------------------------------
+init_per_group(tls1_2, Config) ->
+    setup_tls_server_for_group(tls1_2, Config);
+init_per_group(tls1_3, Config) ->
+    setup_tls_server_for_group(tls1_3, Config);
+init_per_group(revoked_1_2, Config) ->
+    setup_tls_server_for_group(revoked_1_2, Config);
+init_per_group(revoked_1_3, Config) ->
+    setup_tls_server_for_group(revoked_1_3, Config);
+init_per_group(_GroupName, Config) ->
+    Config.
+
+end_per_group(_GroupName, Config) ->
+    Pid = ?config(server_pid, Config),
+    stop_tls_server(Pid),
+    Config.
+
+%%--------------------------------------------------------------------
+init_per_testcase(_TestCase, Config) ->
+    Config.
+
+end_per_testcase(_TestCase, Config) ->
+    Config.
+
+%%--------------------------------------------------------------------
+%% Test Cases --------------------------------------------------------
+%%--------------------------------------------------------------------
+
+ocsp_stapling_without_nonce_and_responder_certs_tls1_2() ->
+    [{doc, "Verify OCSP stapling works without nonce "
+           "and responder certs for tls1.2."}].
+ocsp_stapling_without_nonce_and_responder_certs_tls1_2(Config)
+  when is_list(Config) ->
+    Port = ?config(server_port, Config),
```

Comment:
A general comment, you should use `proplists:get_value/2` instead of the "?config" macro to remove the dependency on the old test_server.

Reply:
done